Fhi.HelseId.Integration.Tests/HelseId.Api/Tests/DPoPTests.cs

```diff
@@ -0,0 +1,99 @@
+using Fhi.ClientCredentialsKeypairs;
+using Fhi.HelseId.Api;
+using Fhi.HelseId.Api.ExtensionMethods;
+using Fhi.HelseId.Integration.Tests.HelseId.Api.Setup;
+using Fhi.HelseId.Integration.Tests.TestFramework;
+using System.Net;
+using System.Text.Json;
+
+namespace Fhi.HelseId.Integration.Tests.HelseId.Api.AppInitiatedTests;
+
+/// <summary>
+/// Need to go through if is as intended
+/// </summary>
+public class DPoPTests 
+{
+    
+    [Test]
+    public async Task GIVEN_AllowDpopTokens_is_true_RequireDPoPTokens_is_false_WHEN_request_contains_dpop_proof_and_validates_ok_THEN_Returns200Ok()
+    {
+        var config = HelseIdApiKonfigurasjonExtensions.CreateHelseIdApiKonfigurasjon(
+            allowDPoPTokens: true, 
+            requireDPoPTokens: false,
+            audience: "fhi:helseid.testing.api",
+            allowedScopes: "fhi:helseid.testing.api/all");
+
+        using var client = CreateDirectHttpClient(config, useDpop: true);
+        var response = await client.GetAsync("api/test");
+        var responseBody = await response.Content.ReadAsStringAsync();
+
+        Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+        Assert.That(responseBody, Is.EqualTo("Hello world!"));
+    }
+    
+
+    [Test]
+    public async Task GIVEN_AllowDpopTokens_is_true_RequireDPoPTokens_is_false_WHEN_no_dpop_proof_and_validates_ok_THEN_Returns200Ok()
+    {
+        var config = HelseIdApiKonfigurasjonExtensions.CreateHelseIdApiKonfigurasjon(
+            allowDPoPTokens: true, 
+            requireDPoPTokens: false,
+            audience: "fhi:helseid.testing.api",
+            allowedScopes: "fhi:helseid.testing.api/all");
+
+        using var client = CreateDirectHttpClient(config, useDpop: false);
+        var response = await client.GetAsync("api/test");
+        var responseBody = await response.Content.ReadAsStringAsync();
+
+        Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+        Assert.That(responseBody, Is.EqualTo("Hello world!"));
+    }
+
+    /// <summary>
+    /// TODO: should use WWW-Authenticate header and check for token not validate
+    /// </summary>
+    /// <returns></returns>
+    [Test]
+    public async Task GIVEN_AllowDpopTokens_is_true_RequireDPoPTokens_is_true_WHEN_no_dpop_proof_and_validates_ok_THEN_Returns401()
+    {
+        var config = HelseIdApiKonfigurasjonExtensions.CreateHelseIdApiKonfigurasjon(
+            allowDPoPTokens: true, 
+            requireDPoPTokens: true,
+            audience: "fhi:helseid.testing.api",
+            allowedScopes: "fhi:helseid.testing.api/all");
+
+        using var client = CreateDirectHttpClient(config, useDpop: false);
```

Comment:
Nit: Trenger vel ikke defaulten her ?  Den settes eksplisitt i all bruk.

Reply:
Jeg gjorde det for tydeliggjøring og lesbarhet i testene på hva forskjellen er.